partiql-eval/src/main/kotlin/org/partiql/eval/internal/Compiler.kt

```diff
@@ -123,8 +131,30 @@ internal class Compiler(
         return ExprPathIndex(root, index)
     }
 
-    // REL
+    @OptIn(PartiQLFunctionExperimental::class)
+    override fun visitRexOpCallStatic(node: Rex.Op.Call.Static, ctx: Unit): Operator {
+        val fn = node.fn.signature
+        val args = node.args.map { visitRex(it, ctx) }
+        val matches = session.functions
+            .flatMap { it.value }
+            .filter { it.signature == fn }
```

Comment:
```suggestion
        val matches = session.functions
            .flatMap { it.value }
            .filterIsInstance<PartiQLFunction.Scalar>
            .filter { it.signature == fn }
```

Then, you can drop the check further below. We know that the `fn.signature` is a signature for a scalar function.